camel/storages/vectordb_storages/qdrant.py

```diff
@@ -0,0 +1,337 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from dataclasses import asdict
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+from qdrant_client import QdrantClient
+from qdrant_client.http.models import (
+    CreateAlias,
+    DeleteAlias,
+    Distance,
+    PointIdsList,
+    PointStruct,
+    UpdateStatus,
+    VectorParams,
+)
+
+from camel.storages.vectordb_storages import (
+    BaseVectorStorage,
+    VectorDBQuery,
+    VectorDBQueryResult,
+    VectorDistance,
+    VectorRecord,
+)
+
+
+class QdrantStorage(BaseVectorStorage):
+    r"""An implementation of the `BaseVectorStorage` for interacting with
+    Qdrant, a vector search engine.
+
+    Args:
+        vector_dim (int): The dimenstion of storing vectors.
+        collection (Optional[str]): Name for the collection in the Qdrant. If
+            not provided, a unique identifier is generated.
+            (default: :obj:`None`)
+        url_and_api_key (Optional[Tuple[str, str]]): Tuple containing the URL
+            and API key for connecting to a remote Qdrant instance.
+            (default: :obj:`None`)
+        path (Optional[str]): Path to a directory for initializing a local
+            Qdrant client. (default: :obj:`None`)
+        distance (VectorDistance): The distance metric for vector comparison
+            (default: :obj:`VectorDistance.DOT`).
+        del_collection (bool): Flag to determine if the collection should be
+            deleted upon object destruction (default: :obj:`False`).
+        **kwargs (Any): Additional keyword arguments for initializing
+            `QdrantClient`.
+
+    Notes:
+        - If `url_and_api_key` is provided, it takes priority and the client
+          will attempt to connect to the remote Qdrant instance using the URL
+          endpoint.
+        - If `url_and_api_key` is not provided and `path` is given, the client
+          will use the local path to initialize Qdrant.
+        - If neither `url_and_api_key` nor `path` is provided, the client will
+          be initialized with an in-memory storage (`":memory:"`).
```

Comment:
add a reference in all docstring like:
References:
https://qdrant.tech/

Reply:
Can we only add it in the class docsting?

Reply:
I think it's also ok, do we have established guidelines or standardized requirements for making references? @lightaime

Reply:
It is commonly used in docstring.